src/Confluent.SchemaRegistry.Serdes.Avro/SpecificSerializerImpl.cs

```diff
@@ -33,23 +34,61 @@ namespace Confluent.SchemaRegistry.Serdes
 {
     internal class SpecificSerializerImpl<T> : IAvroSerializerImpl<T>
     {
+        internal class SpecificSerializerSchemaData
```

Comment:
you use this for non `ISpecificRecord` data as well, so maybe change name to `SerializerSchemaData`.